hnn_core/parallel_backends.py

```diff
@@ -42,11 +45,12 @@ def _clone_and_simulate(net, trial_idx):
     from hnn_core.network_builder import _simulate_single_trial
 
     neuron_net = NetworkBuilder(net, trial_idx=trial_idx)
-    dpl = _simulate_single_trial(neuron_net, trial_idx)
+    times = _simulate_single_trial(neuron_net, trial_idx)
 
     sim_data = neuron_net.get_data_from_neuron()
+    sim_data['times'] = times
```

Comment:
Why not just send `times` back with `get_data_from_neuron()`?

Reply:
We could ... I haven't quite thought about it carefully. My initial thought was that `times` is not a network property but more general to the experimental set up. You could also think of moving `_simulate_single_trial` to be a method of `network_builder` so you can do `network_builder.simulate()`

Reply:
> My initial thought was that `times` is not a network property

I agree, but it's also a little surprising that `_simulate_single_trial` returns `times`, of all things. I'm +1 for keeping it like this though, since networks (even those inside builder?) should be "timeless".

Reply:
Given the fact that `NetworkBuilder` objects return data in time, I don't see how they are timeless (at least after the simulation). That said, I do see an argument for `times` not being considered "data". We can leave it for now.

Reply:
let's address this separately. I think some harmonization of `_simulate_single_trial` and `get_data_from_neuron` may also be necessary when we do that

Reply:
@rythorpe your wish is my command: https://github.com/jonescompneurolab/hnn-core/pull/397/commits/76a7cf9992cc7c301dc1a5c1099de41a69c5fcf2

I did the following:

- remove `get_data_from_neuron` altogether and move it to `_simulate_single_trial`
- move some of the aggregation across ranks inside `aggregate_data`
- remove `_clone_and_simulate` because it just had two statements
- now `_simulate_single_trial` is the workhorse of the simulation and returns all the data. I think it makes more semantic sense to me

The diff of this PR keeps getting large. Please don't make me rebase and merge this PR before the others :)

Reply:
This is beautiful - thanks for humoring me @jasmainak. The code is much easier to follow now!